### PR TITLE
Update Lightning device TOML files to the new schema

### PR DESCRIPTION
**Pull request by astralcai**
**Context:**

This PR is part of the new device capabilities initiative to improve feature parity across the ecosystem. See [this ADR](https://github.com/PennyLaneAI/adrs/pull/78) for more context. A new TOML schema has been defined, and the relevant module implemented in PennyLane:
- https://github.com/PennyLaneAI/pennylane/pull/6407
- https://github.com/PennyLaneAI/pennylane/pull/6433.

As well as updates made to Catalyst:
- https://github.com/PennyLaneAI/catalyst/pull/1275

**Description of the Change:**

- Updates `lightning_qubit.toml`, `lightning_kokkos.toml`, `lightning_gpu.toml` to the new schema
- Removes `_operations` and `_observables` from `LightningQubit`, `LightningKokkos`, and `LightningGPU` as they are now available via `Device.capabilities` that is loaded from the TOML file.

**Benefits:**

A step towards feature parity across the ecosystem.

**Possible Drawbacks:**

- Per discussions when the ADR was developed, `operator.gates.decomp` and `operator.gates.matrix` are removed, and the TOML file no longer prescribes to the framework how an operator should be handled. To ensure consistency of behaviour, this information is temporarily moved to a `_to_matrix_ops` class property to be used by Catalyst, until we have better support for customizable multi-pathway decompositions.

**Related GitHub Issues:**

[sc-71729]

### CODE REVIEW

File: requirements-dev.txt
Comment (astralcai) [inline requirements-dev.txt:21]: TODO: revert this

**Comment by astralcai** [comment]
> ## [Codecov](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI) Report
> Attention: Patch coverage is `36.17021%` with `30 lines` in your changes missing coverage. Please review.
> 
> > Project coverage is 28.37%. Comparing base [(`94577a8`)](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/commit/94577a8f8c7f9f78cb73ffbfff8c919159c10ea1?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI) to head [(`fb96f28`)](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/commit/fb96f28a845847d5676a3a971bef1cd8649dcf19?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI).
> 
> [Files with missing lines](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI)	Patch %	Lines
> [pennylane_lightning/lightning_gpu/lightning_gpu.py](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?src=pr&el=tree&filepath=pennylane_lightning%2Flightning_gpu%2Flightning_gpu.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI#diff-cGVubnlsYW5lX2xpZ2h0bmluZy9saWdodG5pbmdfZ3B1L2xpZ2h0bmluZ19ncHUucHk=)	33.33%	[10 Missing ⚠️ ](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI)
> [...ane_lightning/lightning_kokkos/lightning_kokkos.py](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?src=pr&el=tree&filepath=pennylane_lightning%2Flightning_kokkos%2Flightning_kokkos.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI#diff-cGVubnlsYW5lX2xpZ2h0bmluZy9saWdodG5pbmdfa29ra29zL2xpZ2h0bmluZ19rb2trb3MucHk=)	33.33%	[10 Missing ⚠️ ](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI)
> [...ylane_lightning/lightning_qubit/lightning_qubit.py](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?src=pr&el=tree&filepath=pennylane_lightning%2Flightning_qubit%2Flightning_qubit.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI#diff-cGVubnlsYW5lX2xpZ2h0bmluZy9saWdodG5pbmdfcXViaXQvbGlnaHRuaW5nX3F1Yml0LnB5)	37.50%	[10 Missing ⚠️ ](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI)
> > ❗  There is a different number of reports uploaded between BASE ([94577a8](https://github.com/PennyLaneAI/pennylane-lightning/commit/94577a8f8c7f9f78cb73ffbfff8c919159c10ea1)) and HEAD ([fb96f28](https://github.com/PennyLaneAI/pennylane-lightning/commit/fb96f28a845847d5676a3a971bef1cd8649dcf19)). Click for more details.
> > HEAD has 32 uploads less than BASE
> > 
> > 
> > 
> > Flag
> > BASE ([94577a8](https://github.com/PennyLaneAI/pennylane-lightning/commit/94577a8f8c7f9f78cb73ffbfff8c919159c10ea1))
> > HEAD ([fb96f28](https://github.com/PennyLaneAI/pennylane-lightning/commit/fb96f28a845847d5676a3a971bef1cd8649dcf19))
> > 
> > 
> > 
> > 
> > 
> > 38
> > 6
> 
> Additional details and impacted files
> [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/PennyLaneAI/pennylane-lightning/pull/988?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PennyLaneAI).

This can't be right, I don't know why this is happening 🤷🏻

**Comment by AmintorDusko** [comment]
@astralcai, could you please resolve the conflicts?

**Comment by AmintorDusko** [comment]
Also, could you please add a changelog?

File: .github/CHANGELOG.md
Comment (maliasadi) [inline .github/CHANGELOG.md:32]: Could you add this to the top (line 24)?

File: pennylane_lightning/lightning_gpu/lightning_gpu.py
Comment (maliasadi) [inline pennylane_lightning/lightning_gpu/lightning_gpu.py:95]: What about caching these supported ops and obs methods as protected (private) callables to avoid Python namespace resolutions (i.e., importing `supports_op` `supports_obs` methods from `LightningGPU.capabilities` in multiple places)?

File: pennylane_lightning/lightning_gpu/lightning_gpu.py
Comment (maliasadi) [inline pennylane_lightning/lightning_gpu/lightning_gpu.py:147]: Shall we prefer the compact version in PL?
```suggestion
    if isinstance(op, (Conditional, MidMeasureMP, PauliRot)): return False
```

File: pennylane_lightning/lightning_gpu/lightning_gpu.py
Comment (maliasadi) [inline pennylane_lightning/lightning_gpu/lightning_gpu.py:150]:
```suggestion
    if not qml.operation.is_trainable(op): return True
```

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml
Comment (maliasadi) [inline pennylane_lightning/lightning_qubit/lightning_qubit.toml:75]: This is natively (C++) supported in Lightning

**Review by maliasadi — COMMENTED** [review]
Nice work @astralcai! Happy to approve once the suggestions are resolved.

File: pennylane_lightning/lightning_gpu/lightning_gpu.py
Comment (astralcai) [reply pennylane_lightning/lightning_gpu/lightning_gpu.py:78]: This mirrors the `operators.gates.matrix` from the original TOML file. I'm not sure why it wasn't included in the first place but I decided not to change it.

File: pennylane_lightning/lightning_gpu/lightning_gpu.py
Comment (astralcai) [reply pennylane_lightning/lightning_gpu/lightning_gpu.py:95]: What do you mean? I'm not very familiar with namespace resolution.

File: pennylane_lightning/lightning_gpu/lightning_gpu.py
Comment (astralcai) [reply pennylane_lightning/lightning_gpu/lightning_gpu.py:147]: Will update.

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml
Comment (astralcai) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml:75]: It was previously in `operators.gates.decomp`, which means that Catalyst was explicitly told to decompose this. Do we want to change that?

File: pennylane_lightning/lightning_gpu/lightning_gpu.py
Comment (maliasadi) [reply pennylane_lightning/lightning_gpu/lightning_gpu.py:78]: Okay -- I'll add a SC ticket for checking this in a follow-up PR! It would be easy to fix after gathering some benchmark.

File: pennylane_lightning/lightning_gpu/lightning_gpu.py
Comment (maliasadi) [reply pennylane_lightning/lightning_gpu/lightning_gpu.py]: Thanks!

File: pennylane_lightning/lightning_qubit/lightning_qubit.toml
Comment (maliasadi) [reply pennylane_lightning/lightning_qubit/lightning_qubit.toml:75]: Hmm.. okay -- there isn't any frontend support in Catalyst for MCX. It should be easy to fix as a follow-up.

**Review by AmintorDusko — DISMISSED** [review]
Looks good! No blocker from me!

**Comment by astralcai** [comment]
@AmintorDusko The doc build is failing because it's building with latest PL, which still doesn't contain the changes that this PR depends on. Once we merge the PR into PL, it will unblock the doc build for Lightning.

**Comment by AmintorDusko** [comment]
> @AmintorDusko The doc build is failing because it's building with latest PL, which still doesn't contain the changes that this PR depends on. Once we merge the PR into PL, it will unblock the doc build for Lightning.

Ohh. Thank you for clarifying, Good job here!

**Review by AmintorDusko — APPROVED** [review]

**Review by maliasadi — APPROVED** [review]
Thanks @astralcai! Feel free to merge this after CIs are completed :beach_umbrella:

**Pull request merged**
